vite.config.ts

```diff
@@ -0,0 +1,28 @@
+import { defineConfig } from "vite";
+import path from "node:path";
+import dts from "vite-plugin-dts";
+
+// https://vitejs.dev/config/
```

Comment:
what are we using this config for?

Reply:
this is the config for vite to build the packages and type definitions

Reply:
But where do we use vite here?

Reply:
ah, it's a little roundabout - vite is called by `npm run build` (https://github.com/viamrobotics/viam-typescript-sdk/blob/b1abc17a7a96bcfe1dbfd4e60d5220cef9de676d/package.json#L15), which is called by `make build`

Reply:
Vite will generate that nice UMD + ES module combo